tools/topology/CMakeLists.txt

```diff
@@ -6,68 +6,72 @@ file(GLOB TPLG_DEPS
 	sof/*.m4
 )
 
-set(MACHINES
-	sof-cht-nocodec
-	sof-cht-max98090
-	sof-cht-rt5682
-	sof-hda-intel-common
-	sof-apl-nocodec
-	sof-apl-keyword-detect
-	sof-byt-nocodec
-	sof-bdw-rt286
-	sof-bdw-rt5640
-	sof-byt-rt5640
-	sof-byt-rt5645
-	sof-byt-rt5651
-	sof-byt-da7213
-	sof-cnl-rt274
-	sof-hsw-rt5640
-	sof-apl-tdf8532
-	sof-apl-pcm512x
-	sof-apl-pcm512x-nohdmi
-	sof-apl-rt298
-	sof-apl-wm8804
-	sof-apl-da7219
-	sof-glk-da7219
-	sof-glk-rt5682
-	sof-icl-nocodec
-	sof-apl-eq-pcm512x
-	sof-apl-eq-dmic
-	sof-apl-dmic-2ch
-	sof-apl-dmic-4ch
-	sof-icl-dmic-4ch
-	sof-apl-src-pcm512x
-	sof-cml-rt5682
-	sof-whl-rt5682
-	sof-cml-src-rt5682
-	sof-cnl-nocodec
+set(TPLGS
+	"sof-cht-nocodec\;sof-cht-nocodec\; "
+	"sof-cht-max98090\;sof-cht-max98090\; "
+	"sof-cht-rt5682\;sof-cht-rt5682\; "
+	"sof-hda-intel-common\;sof-hda-intel-common\; "
+	"sof-apl-nocodec\;sof-apl-nocodec\; "
+	"sof-apl-keyword-detect\;sof-apl-keyword-detect\; "
+	"sof-byt-nocodec\;sof-byt-nocodec\; "
+	"sof-bdw-rt286\;sof-bdw-rt286\; "
+	"sof-bdw-rt5640\;sof-bdw-rt5640\; "
+	"sof-byt-rt5640\;sof-byt-rt5640\; "
+	"sof-byt-rt5645\;sof-byt-rt5645\; "
+	"sof-byt-rt5651\;sof-byt-rt5651\; "
+	"sof-byt-da7213\;sof-byt-da7213\; "
+	"sof-cnl-rt274\;sof-cnl-rt274\; "
+	"sof-hsw-rt5640\;sof-hsw-rt5640\; "
+	"sof-apl-tdf8532\;sof-apl-tdf8532\; "
+	"sof-apl-pcm512x\;sof-apl-pcm512x\; "
+	"sof-apl-pcm512x-nohdmi\;sof-apl-pcm512x-nohdmi\; "
+	"sof-apl-rt298\;sof-apl-rt298\; "
+	"sof-apl-wm8804\;sof-apl-wm8804\; "
+	"sof-apl-da7219\;sof-apl-da7219\; "
+	"sof-glk-da7219\;sof-glk-da7219\; "
+	"sof-glk-rt5682\;sof-glk-rt5682\; "
+	"sof-icl-nocodec\;sof-icl-nocodec\; "
+	"sof-apl-eq-pcm512x\;sof-apl-eq-pcm512x\; "
+	"sof-apl-eq-dmic\;sof-apl-eq-dmic\; "
+	"sof-apl-dmic-2ch\;sof-apl-dmic-2ch\; "
+	"sof-apl-dmic-4ch\;sof-apl-dmic-4ch\; "
+	"sof-icl-dmic-4ch\;sof-icl-dmic-4ch\; "
+	"sof-apl-src-pcm512x\;sof-apl-src-pcm512x\; "
+	"sof-cml-rt5682\;sof-cml-rt5682\;PLATFORM=cml"
+	"sof-cml-rt5682\;sof-whl-rt5682\;PLATFORM=whl"
+	"sof-cml-src-rt5682\;sof-cml-src-rt5682\; "
+	"sof-cnl-nocodec\;sof-cnl-nocodec\; "
 )
 
 add_custom_target(topologies ALL)
 
-foreach(machine ${MACHINES})
+foreach(tplg ${TPLGS})
+	list(GET tplg 0 machine)
+	list(GET tplg 1 output)
+	list(GET tplg 2 custom_defines)
```

Comment:
I think it should be singular form `custom_define` to not encourage to use multiple defines that will not work, afaik `-D` in m4 needs 0 o 1 defines, when you do something like `-DA=1 B=2` it will treat B=2 as file name

Reply:
@jajanusz yes you are right. m4 does require a -D for every individual define. will fix this to make it generic for multiple defines.

Reply:
@jajanusz @lgirdwood fixed now. we can now have multiple defines for the same target